vault/quotas/quotas_rate_limit.go

```diff
@@ -92,6 +92,20 @@ func NewRateLimitQuota(name, nsPath, mountPath string, rate float64, interval, b
 	}
 }
 
+func (q *RateLimitQuota) Clone() *RateLimitQuota {
```

Comment:
@alexanderbez Should we push the clone() function into the quota interface and potentially call this in setQuotaLocked function before the initialize function? This way even the lease count quota can implement clone and not run into similar issues.